pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/api/go/profile"
+	"github.com/lf-edge/eve/pkg/pillar/flextimer"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+const (
+	defaultLocalProfileServerPort = "8888"
+	savedLocalProfileFile         = "lastlocalprofile"
+)
+
+// urlAndSrcIP structure for mapping source IP and url of local server
+type urlAndSrcIP struct {
+	srcIP     net.IP
+	actualURL string
+}
+
+func getLocalProfileURL(localProfileServer string) (string, error) {
+	localProfileURL := fmt.Sprintf("http://%s", localProfileServer)
+	u, err := url.Parse(localProfileURL)
+	if err != nil {
+		return "", fmt.Errorf("url.Parse: %s", err)
+	}
+	if u.Port() == "" {
+		localProfileURL = fmt.Sprintf("%s:%s", localProfileURL, defaultLocalProfileServerPort)
+	}
+	return fmt.Sprintf("%s/api/v1/local_profile", localProfileURL), nil
+}
+
+// Run a periodic fetch of the currentProfile
+func localProfileTimerTask(handleChannel chan interface{}, getconfigCtx *getconfigContext) {
+
+	ctx := getconfigCtx.zedagentCtx
+	iteration := 0
+
+	// use ConfigInterval as localProfileInterval
+	localProfileInterval := ctx.globalConfig.GlobalValueInt(types.ConfigInterval)
+	interval := time.Duration(localProfileInterval) * time.Second
+	max := float64(interval)
+	min := max * 0.3
+	ticker := flextimer.NewRangeTicker(time.Duration(min),
+		time.Duration(max))
+	// Return handle to caller
+	handleChannel <- ticker
+
+	wdName := agentName + "currentProfile"
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	ctx.ps.RegisterFileWatchdog(wdName)
+
```

Comment:
Here we should have a 
msg, err := readSavedLocalProfile()
if err {
log
} else if msg != nil {
x, err := parseLocalProfile()
log "starting with localProfile X"
ctx.localProfile = X
}

validated(msg, skip